sdks/ios/ZelloChannelKit/ZelloChannelKit/network/ZCCSocket.m

```diff
@@ -818,11 +830,36 @@ - (void)handleImage:(NSDictionary *)encoded original:(NSString *)original {
   }];
 }
 
+/**
+ * Report an error parsing a JSON message from the server
+ *
+ * @param message the type of message
+ * @param key the key we were trying to retrieve
+ * @param description a description of the error encountered
+ * @param original the original string we were trying to parse
+ */
+- (void)reportInvalidJSONInMessage:(nullable NSString *)message key:(NSString *)key errorDescription:(NSString *)description original:(NSString *)original {
+  id<ZCCSocketDelegate> delegate = self.delegate;
+  if ([delegate respondsToSelector:@selector(socket:didEncounterErrorParsingMessage:)]) {
+    NSMutableDictionary *info = [@{ZCCServerInvalidMessageKey:original,
+                                  ZCCInvalidJSONKeyKey:key,
+                                  ZCCInvalidJSONProblemKey:description} mutableCopy];
+    if (message) {
+      info[ZCCInvalidJSONMessageKey] = message;
+    }
+    NSError *error = [NSError errorWithDomain:ZCCErrorDomain code:ZCCErrorCodeInvalidMessage userInfo:info];
+    [self.delegateRunner runAsync:^{
+      [delegate socket:self didEncounterErrorParsingMessage:error];
+    }];
```

Comment:
Just an observatoion, this particular block seems to be a little repetitive between here `reportInvalidJSONInMessage` and `reportInvalidStringMessage`. May want to factor out of there are more of the same.

Reply:
They have some shared structure, but I don't think there's enough repeated to make it worth refactoring at this point.